arm64-assembly-optimization.md

```diff
@@ -698,17 +698,90 @@ Another way to use efficient instructions is to consult the [software
 optimization guide](README.md#building-for-graviton2-graviton3-and-graviton3e).
 Many different combinations of instructions can accomplish the same result, and
 some are obviously better than others. Some instructions, like the
-absolute-difference-accumulate-long instructions (`SABAL` and `UABAL`) can only
+absolute-difference-accumulate-long instructions (`sabal` and `uabal`) can only
```

Comment:
if these can be executed only on half the pipelines, and the same functionality is achieved with the other instructions that utilize all the pipes, when do we use these instructions at all

Reply:
Excellent question!

There is a trade off, of course. As you can see in the example, it takes more instructions to avoid using `uabal` because you have to compute the accumulation another way. It works well in this example because we can parallelize the SAD computation across 64 bytes. If the amount of loop unrolling was more limited (maybe because a result from the previous iteration is needed in the next), we might not be able to parallelize enough SAD computations to keep 4 execution units full. In that case it would probably make sense to use 'uabal` since we would already be unable to fill up the queues of 4 execution units.

I will add this text to the document as well!

Reply:
Let me know if this explanation is clear.

Reply:
yes, it makes sense, thanks for adding it to the doc.